File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.py
n, m = map(int, input().split())    # 수의 개수, 수의 합 m
numbers = list(map(int, input().split()))

l = 0
r = 1
result = 0
while r <= n and l <= r:
    if sum(numbers[l:r]) == m:
        result += 1
        r += 1
    elif sum(numbers[l:r]) < m:
        r += 1
    else:
        l += 1

print(result)

File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

let [n, m] = input[0].split(" ").map(Number);
let arr = input[1].split(" ").map(Number);
let result = 0;

for (let i = 0; i < n; i++) {
  let intervalSum = 0;
  let j = i;
  while (j < n) {
    intervalSum += arr[j];
    j++;
    if (intervalSum == m) {
      result++;
    } else if (intervalSum > m) break;
  }
}
console.log(result);
